### DISCUSSION TITLE
Add to Readme: updated instructions on how to setup with macOS homebrew

### DISCUSSION THREAD

**Discussion by robrecord** [wee-slack/wee-slack#881] [Ideas]
Hi, the instructions on how to install this on macOS with homebrew were inadequate for me.

Since python2 has long since left the scene, we could remove the mention of python2. I also had big problems with weechat scripts loading, and a confusing python version situation.

## Correct websocket-client install

Since weechat seems to require a specific version of python (as far as I can tell?) which for me was python 3.11 - when my system was 3.9 and my installed homebrew version was 3.10, it made the process of installing the websocket client very unintuitive.

My instructions would be:

```
brew install python@3.11
brew link python@3.11
python3.11 -m ensurepip
python3.11 -m pip install websocket-client
```

## Fix for scripts not loading

I also had to add the following workaround to my .zshrc (or your .bashrc), otherwise the scripts would not load:

```
export WEECHAT_HOME=~/.weechat
export OBJC_DISABLE_INITIALIZE_FORK_SAFETY=YES
```

Then continue the rest of the installation.

**Comment by trygveaa** [comment]
Hi. Thanks for the info. I'm not using macOS myself, so can't keep those instructions up to date. I have a couple of questions to your commands.

> Since weechat seems to require a specific version of python (as far as I can tell?) which for me was python 3.11 - when my system was 3.9 and my installed homebrew version was 3.10, it made the process of installing the websocket client very unintuitive.
> 
> My instructions would be:
> 
> ```
> brew install python@3.11
> brew link python@3.11
> python3.11 -m ensurepip
> python3.11 -m pip install websocket-client
> ```

Wouldn't Python 3.11 have been installed when you installed WeeChat? Is it necessary to run `brew install python@3.11`?

> I also had to add the following workaround to my .zshrc (or your .bashrc), otherwise the scripts would not load:
> 
> ```
> export WEECHAT_HOME=~/.weechat
> export OBJC_DISABLE_INITIALIZE_FORK_SAFETY=YES
> ```

Right, this is a [known WeeChat issue](https://github.com/weechat/weechat/issues/1723). Is it necessary to set `WEECHAT_HOME` though? I see people mentioning it in that thread as well, but I don't see why it would be related to the problem. Including that in the instructions might be problematic for users that use XDG directories.

**Reply by robrecord** [reply]
Thank you for considering this. 

Yes I think you're right; i think for me python 3.11 wasn't in my path since it appeared to be installed by Weechat in its own special place; I could be wrong - I'd need to wipe clean and recreate the situation to check this. 

Perhaps the right instructions would include a way to see which version of python (and path to the binary) is in use by weechat, and to invoke this path when running the pip commands. Then the install commands would not be necessary.

Regarding the home directory: you're absolutely right; the correct home is `~/.config/weechat` and that does work by default; no need to set the home as above. Especially as I was using XDG home and the config was then being excluded from my dotfiles.

**Reply by trygveaa** [reply]
> Yes I think you're right; i think for me python 3.11 wasn't in my path since it appeared to be installed by Weechat in its own special place; I could be wrong - I'd need to wipe clean and recreate the situation to check this.
> 
> Perhaps the right instructions would include a way to see which version of python (and path to the binary) is in use by weechat, and to invoke this path when running the pip commands. Then the install commands would not be necessary.

Yeah, as far as I understand, the `brew link python@3.11` was what made it available in path. So the user would have to either run this, or to find where this python is installed. Does `brew info python@3.11` show this?
